cmd/api/app/routes/metrics/handler.go

```diff
@@ -0,0 +1,68 @@
+package metrics
+
+import (
+ 
+	"net/http"
+	 
+	"github.com/karmada-io/dashboard/cmd/metrics-scraper/app/scrape"
+	"github.com/gin-gonic/gin"
+	"github.com/karmada-io/dashboard/cmd/api/app/router"
+ 
+)
+ 
+var requests = make(chan scrape.SaveRequest)
+
+func getMetrics(c *gin.Context) {
+	appName := c.Param("app_name")
+	queryType := c.Query("type")
+
+	if queryType == "sync_on" || queryType == "sync_off" {
+		syncValue := 0
+		if queryType == "sync_on" {
+			syncValue = 1
+		}
+		scrape.HandleSyncOperation(c, appName, syncValue, queryType)
+		return
+	}
+
+	if queryType == "metricsdetails" {
+		queryMetrics(c)
+		return
+	}
+
+	if queryType == "sync_status" {
+		scrape.CheckAppStatus(c)
+		return
+	}
+
+	allMetrics, errors, err := scrape.FetchMetrics(c.Request.Context(), appName, requests)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"errors": errors, "error": err.Error()})
+		return
+	}
+	if len(allMetrics) > 0 {
+		c.JSON(http.StatusOK, allMetrics)
+	} else {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "No metrics data found", "errors": errors})
+	}
+}
+ 
+func init() {
+	go scrape.InitDatabase()
```

Comment:
here the init process of database can be moved to `metricsscraper.go` file
![image](https://github.com/user-attachments/assets/80a6bdbe-5c1c-4e44-b86a-3f0bf5d51b19)